crates/bevy_ecs/src/schedule/mod.rs

```diff
@@ -57,6 +57,10 @@ impl Schedule {
     }
 
     pub fn add_stage<S: Stage>(&mut self, name: &str, stage: S) -> &mut Self {
+        if self.stages.get(name).is_some() {
+            panic!("Stage already exists: {}.", name);
+        }
+
         self.stage_order.push(name.to_string());
         self.stages.insert(name.to_string(), Box::new(stage));
```

Comment:
```suggestion
        self.stage_order.push(name.to_string());
        let prev = self.stages.insert(name.to_string(), Box::new(stage));
        if prev.is_some() {
            panic!("Stage already exists: {}.", name);
        }
```

Reply:
You still get duplicate Stage-names in the `stage_order`-Vec, and remove all the systems belonging to the old Stage, and THEN panic.

Instead of directly panicing without doing that extra amount of (buggy) work

Reply:
A panic will abort the program anyway so it doesn't matter that this bit of extra work is being done when the user made an error. It does however make program startup a bit faster when there are no duplicate stages as it doesn't have to do a duplicate hash table lookup.

Reply:
I see what you mean and apologize

If panicking stays as the preferred way to exit this function, `add_stage_before` and `_after` would also benefit from this change of order. In the case of Error-propagation this would, however, introduce the bugs mentioned above. 

Therefore, I'll wait with committing this change until a decision is made on whether panicking or error-propagation is to be used.

Reply:
I do want to push for more error handling where it makes sense, but (at least in AppBuilder methods) I think im ok with panic-ing in the interest of ergonomics.

1. If we return a Result, then i expect 99% of cases to look like this:
    ```rust
    app
        .add_stage("my_stage", SystemStage::default())
        .unwrap()
        .add_system_to_stage("my_stage", foo.system())
    ```
2. Given that we are likely moving to ZST-style labels for systems, I can see a world where we adopt them for stages too. This would significantly reduce the likelihood of collisions across plugins.
3. We could consider doing error handling in Schedule and unwrapping in AppBuilder, but even that has "app building ergonomics" implications, because people can interact with Schedules directly when building apps.

I think "app builder error handling" is a conversation to be had in a separate issue/pr because thats a longer conversation (that is totally worth having).

So for now, lets assume panicing is the way to go here.